mkt/websites/management/commands/import_websites_from_csv.py

```diff
@@ -202,8 +214,65 @@ def parse(self, filename):
         except IOError as err:
             raise CommandError(err)
 
+    def assign_popularity(self):
+        print 'Setting regional popularity values...'
+        for region in self.ranking_per_region.keys():
+            websites_len = len(self.ranking_per_region[region])
+            print u'Setting regional popularity for %d site(s) in %s' % (
+                websites_len, unicode(REGIONS_CHOICES_ID_DICT[region].name))
+            # Sort sites by rank in that region.
+            websites = sorted(self.ranking_per_region[region],
+                              key=itemgetter(1), reverse=True)
+            # Take the same number of popularity values for apps in that
+            # region.
+            apps_popularity = (Installs.objects.filter(region=region)
+                                       .values_list('value', flat=True)
+                                       .order_by('-value')[:websites_len])
+
+            for i, app_popularity_value in enumerate(apps_popularity):
+                # Steal popularity value, minus one just to get a chance to end
+                # up with a more stable ordering (no equal values).
+                pk = websites[i][0]
+                popularity, created = WebsitePopularity.objects.get_or_create(
+                    website_id=pk, region=region)
+                popularity.update(value=app_popularity_value - 1)
+        print 'Setting global popularity values...'
+        GLOBAL_REGION = 0
+        for pk in self.websites:
+            values = list(WebsitePopularity.objects
+                                           .filter(website=pk)
+                                           .exclude(region=GLOBAL_REGION)
+                                           .values_list('value', flat=True))
+            popularity, created = WebsitePopularity.objects.get_or_create(
+                website_id=pk, region=GLOBAL_REGION)
+            popularity.update(value=sum(values))
```

Comment:
I had to read this a few times to understand everything, sorry for the review delay.